### PR TITLE
Rename build steps and spike unity prop and gov/sudo testing CI step

### PR DESCRIPTION
**Pull request by the-frey**
This:

- Updates the names of a bunch of CI so it's clearer what it does
- Adds a unity contract CI runner to ensure there are no regressions
- Adds a unity contract governance action to ensure there are no regressions in Juno gov routing to contracts
- Adds a separate docker build _but not push_ step for PRs only (to sense check builds succeed)

### CODE REVIEW

**Review by joeabbey — APPROVED** [review]
Nice, preserving unity ... good

**Comment by faddat** [comment]
reading the code now, but I really love the way you're approaching this

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-name: ci
+name: Build and Push Docker Image
```
Comment (faddat) [inline .github/workflows/docker.yml:1]: can't tell you how many times I've meant to fix naming in ci.  Thanks!

**Review by JakeHartnell — APPROVED** [review]

**Comment by j0nl1** [comment]
Dockerfile don't use arch arg anymore and it would be good include arm64 support in the buildx

**Pull request merged**